www/js/controllers.js

```diff
@@ -102,86 +88,126 @@ angular.module('starter.controllers', [])
       };
-
     });
   }
 
   // where the map is initiated and called
   navigator.geolocation.getCurrentPosition($scope.drawMap);
-  navigator.geolocation.getCurrentPosition(function(position){
-    $scope.coordinates = {
-      lat: position.coords.latitude, 
-      lng: position.coords.longitude
+
+  $scope.sendCenterLocation = function(){
+    var lat = $scope.center_coords ? $scope.center_coords.G : $scope.myLocation.lat;
+    var lng = $scope.center_coords ? $scope.center_coords.K : $scope.myLocation.lng;
+    sendPickupRequest(lat, lng);
+  }
+
+  function sendPickupRequest(lat, lng){
+    var request = {
+      request: {
+        latitude: lat,
+        longitude: lng
+      }
     };
-  });
 
-  $scope.sendCurrentLocation = function(){
-    console.log($scope.coordinates);
-    var geocoder = new google.maps.Geocoder;
-    geocoder.geocode({'location': $scope.coordinates}, function(results, status){
-      console.log(results);
+    $ionicLoading.show({
+      templateUrl: 'templates/notification/waiting_request_page.html',
+      scope: $scope
     });
-    // send out http put request to location
+
+    $http.post(CtrlService.urlFactory('users/'+ CtrlService.currentUser.id +'/requests'), request)
+    .then(function(response){
+      $scope.LastRequestId = response.data.id;
+      $scope.requestMade = true;
+      console.log($scope.LastRequestId);
+    })
+    .catch(function(response){
+      console.log(response);
+      $ionicLoading.hide();
+      $ionicPopup.alert({
+        title: 'There was an error in your request.',
+        template: 'Please try again later.'
+      });
+    })
   }
 
-  $scope.sendPickupLocation = function(query){
-    // console.log(query);
-    var geocoder = new google.maps.Geocoder;
-    geocoder.geocode({'address': query + ', Hong Kong'}, function(results, status){
-      console.log('query----->', results[0].formatted_address, 'coords----->' ,results[0].geometry.location);
-      $scope.query_coords = [results[0].geometry.location.G, results[0].geometry.location.K];
-      console.log($scope.query_coords);
-    });
+  $scope.cancelRequest = function(){
+    $http.put(CtrlService.urlFactory('requests/'+ $scope.LastRequestId + '/cancel_request'))
+    .then(function(response){
+      console.log(response);
+      $ionicLoading.hide();
+      $scope.requestMade = false;
+    })
+    .catch(function(response){
+      console.log(response);
+    })
+  }
+})
+
+.controller('profileCtrl', function(CtrlService, $scope, $auth, $http, $window, $state) {
+
+  $scope.personalForm = {};
+  $scope.carForm      = {};
+
+  $scope.editPersonalInfo = function(form){
+    // check for nulls
+    console.log(form);
   }
 
-});
+  $scope.editCarInfo = function(form){
+    console.log(form);
+  }
+
+  function editUserRequest(formType){
+    $http.put(CtrlService.urlFactory('users/')+ CtrlService.currentUser, editForm)
+  }
+})
+
+.service('CtrlService', function(){
```

Comment:
My opinion is that it is easier to access service attributes this way. Just preference, I guess. Unless there is a real disadvantage to using services. 

Also, I'm not a big fan of using JSON.parse($window.localStorage.getItem.('currentUser') every time I need a superclass variable. I may include that later as a method to call from the service as well.